### ISSUE TITLE
Harmonize standard python config variable names 

### ISSUE THREAD

**Issue by bryngemark** [LDMXAnalysis/ldmx-sw#1354]
**Is your feature request related to a problem? Please describe.**
_Ex. I'm always frustrated when [...]_
... I try to set something basic like the input collection or input pass name to use in my config on the fly, and I have to look up what exact naming scheme that particular producer uses for it. `input_pass_name`, `digi_pass_name`, `input_hit_pass_name`, or maybe `inputPassName`?

**Describe the solution you'd like**
I would like to have one standard way of defining this type of basic variable that almost every producer uses, and just harmonize it. 

**Describe alternatives you've considered**
I'm wondering if there are good examples of when we need multiple input collections (and associated passes)? I think the `EcalRecHitProducer` is one of the very few I've encountered where there are multiple and I suppose we should have a clever solution there. _Alternatively_, we'd live with looking up the proper name for these few cases.

**Comment by tvami** [comment]
If I look at `pass_name` in the python files: https://github.com/search?q=repo%3ALDMX-Software%2Fldmx-sw+pass_name++language%3APython&type=code

it seems it's mostly the `input_pass_name` that's used, and then `inputPassName` is used in the TS. 
Based on this I'd say we should go with `input_pass_name` to minimize work on the transition.

I think generally it would be good to have some guidelines on the code/naming style. Like in the cxx the same variable should be called with camel, i.e. having the `inputPassName`. And then also enforce that the global variables have the trailing underscore (so maybe that example is actually `inputPassName_`)

**Comment by bryngemark** [comment]
@tvami i think these are good suggestions, and if we find a general principle, it will be easier for streamlining input and output collection variable names etc too.

**Comment by tomeichlersmith** [comment]
While it would be a big lift, [`snake_case`](https://google.github.io/styleguide/cppguide.html#Variable_Names) is actually what the Google C++ style guide suggests for variable names. (I'm guessing `clang-format` makes the choice to not check the case of variable names when formatting which is why this has been missed.) This would imply that we should change the C++ names to be snake case and then the Python config names and the C++ member variables would only differ by a trailing underscore.

_Edit_: `clang-format` [limits itself to local changes that are irrelevant to the compiler](https://stackoverflow.com/questions/73788989/how-to-configure-naming-conventions-with-clang-format), so it does not check for variable names (among other things). This makes its Google style a subset of the Google style guide linked above.

**Comment by bryngemark** [comment]
How are we doing on this at this point?

**Comment by tomeichlersmith** [comment]
It has kinda stalled since I am unaware of a tool that can algorithmically check it for us. (Maybe a Python linter like ruff?)

In any case, even if we have a tool to check, we'd still need to copy the name changes from the Python config to the C++ where the `get` call is made.

**Comment by tvami** [comment]
> How are we doing on this at this point?

I think the issue is that some processors take more than one object. So to allow each to have a different pass names (*), we cant anymore unify to `input_pass_name` as we discussed earlier. Beyond that I think we are pretty good, looking at https://github.com/search?q=repo%3ALDMX-Software%2Fldmx-sw+pass_name++language%3APython&type=code again, I see `input_pass_name`, or the `<object>_pass_name` method like `sp_pass_name` and `rec_pass_name`. So with that I think we can close this issue, and the only thing left is to ensure that we dont use variables that dont exists, which is what the issue https://github.com/LDMX-Software/ldmx-sw/issues/1303 is about

(*) the need for this, I actually run into: when we resimmed the ecal PN sample the scoring plan needed a different pass name than the rest, so we need to maintain this functionality

**Comment by bryngemark** [comment]
yes i agree. flexibility in pass name for every input collection is important. but i think we have come as far as reasonably possible here with case harmonization etc. well done :) closing this issue.
